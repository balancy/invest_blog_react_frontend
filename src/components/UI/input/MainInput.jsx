import React from 'react'

const MainInput = (props) => {
    return (
        <input className="MainInput" {...props}/>
    )
}

export default MainInput