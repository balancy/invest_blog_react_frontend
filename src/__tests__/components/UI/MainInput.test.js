import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MainInput from '../../../components/UI/input/MainInput';


test('should render MainInput component', () => {
  render(
    <MainInput
      data-testid='mainInput-1'
      placeholder='test'
      defaultValue='5'
    />
  )
  const mainInput = screen.getByTestId('mainInput-1')

  expect(mainInput).toBeEnabled()
  expect(mainInput).toBeInstanceOf(HTMLInputElement)
  expect(mainInput).toBeInTheDocument()
  expect(mainInput).toContainHTML('input')
  expect(mainInput).toHaveAttribute('placeholder')
  expect(mainInput).toHaveClass('MainInput')
  expect(mainInput).toHaveValue('5')
})


test('matches snapshot', () => {
  const tree = renderer
    .create(
      <MainInput
        placeholder='test'
        defaultValue='5'
      />
    ).toJSON()
  expect(tree).toMatchSnapshot()
})