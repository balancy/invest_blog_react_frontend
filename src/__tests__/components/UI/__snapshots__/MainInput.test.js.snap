// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  className="MainInput"
  defaultValue="5"
  placeholder="test"
/>
`;
